pydoctor/model.py

```diff
@@ -501,7 +519,47 @@ def setup(self) -> None:
         super().setup()
         self.rawbases: List[str] = []
         self.subclasses: List[Class] = []
+        self._mro: Optional[List[Union['Class', str]]] = None
+    
+    def init_mro(self) -> None:
+        """
+        Compute the correct value of the method resolution order returned by L{mro()}.
+        """
+        try:
+            self._mro = compute_mro(self)
+        except ValueError as e:
+            self.report(str(e), 'mro')
+            self._mro = list(self.allbases(True))
+
+    @overload
+    def mro(self, include_external:'Literal[True]'=True) -> List[Union['Class', str]]:...
+    @overload
+    def mro(self, include_external:'Literal[False]') -> List['Class']:...
+    def mro(self, include_external:bool=True) -> List[Union['Class', str]]: # type:ignore[misc]
+        """
+        Get the method resution order of this class. 
+
+        @note: The actual correct value is only set in post-processing, if L{mro()} is called
+            in the AST visitors, it will return the same as C{list(self.allbases(True))}.
+        """
+        if self._mro is None:
+            return list(self.allbases(True))
+        if include_external is False:
+            return [o for o in self._mro if isinstance(o, Class)]
+        else:
+            return self._mro
 
+    @property
+    def bases(self) -> List[str]:
+        return [self.parent.expandName(str_base)for str_base in self.rawbases]
+    
+    @property
+    def baseobjects(self) -> List[Optional['Class']]:
```

Comment:
This is not the right way to handle baseobjects attribute, we should compute it only once, in post processing. But there is probably code that relies on some parts of the name resolving which relied on this attribute. So maybe we’ll have to compte it twice once at the time of visiting the classdef and in post processing.